### ISSUE TITLE
[QA] Fix clippy identity_op warning in decoder.rs

### ISSUE THREAD

**Issue by drizki** [drizki/voxclaw#330]
## Problem
Clippy reports an `identity_op` warning in `src/stt/whisper/inference/decoder.rs:56`:

```
warning: this operation has no effect
  --> src/stt/whisper/inference/decoder.rs:56:45
   |
56 |     let dummy_kv_data: Vec<f32> = vec![0.0f32; num_heads * 1 * head_dim];
   |                                                ^^^^^^^^^^^^^ help: consider reducing it to: `num_heads`
```

## Fix
Replace `num_heads * 1 * head_dim` with `num_heads * head_dim` (the `* 1` is a no-op, likely a placeholder for sequence length).

Add a comment explaining the dimension: `num_heads * seq_len * head_dim` where `seq_len = 1` for single-step decoding, if that's the intent.

## Files
- `src/stt/whisper/inference/decoder.rs:56`

**Comment by drizki** [comment]
This warning no longer exists in the codebase. The `decoder.rs` file was refactored in #276 and the offending code with `num_heads * 1 * head_dim` was removed during that restructuring.

Verified:
- `cargo clippy` reports 0 warnings
- The pattern `* 1 *` no longer appears anywhere in the source
- Line 56 of `decoder.rs` is now different code entirely

Closing as obsolete.
